network/src/lib.rs

```diff
@@ -181,6 +192,11 @@ impl Node {
     async fn process_incoming_messages(&self, receiver: &mut Receiver) -> Result<(), NetworkError> {
         let Message { direction, payload } = receiver.recv().await.ok_or(NetworkError::ReceiverFailedToParse)?;
 
+        if self.environment.is_bootnode() && !(payload == Payload::GetPeers || direction == Direction::Internal) {
+            // the bootstrapper nodes should ignore inbound messages other than GetPeers
+            return Ok(());
```

Comment:
We might want to introduce a new `NetworkError::TooManyConnections`  or `::ConnectionLimitBreach` in order to indicate that this operation failed; it doesn't matter during regular runtime, but will for e.g. a RPC to connect to a specific address or in tests.

Reply:
(this was of course directed at `initialize_connection`, which got amended now :ok_hand:)